spring-kafka/src/main/java/org/springframework/kafka/config/StreamsBuilderFactoryBean.java

```diff
@@ -337,6 +337,12 @@ protected StreamsBuilder createInstance() {
 
 	@Override
 	public boolean isAutoStartup() {
+		try {
+			this.topology = getObject().build(this.properties);
+		}
+		catch (Exception e) {
+			throw new RuntimeException(e);
+		}
```

Comment:
It might be better to add this to the `afterPropertiesSet()` method instead of here. Doing it here adds more responsibilities to the `isAutoStartup()` method, which should technically only return the current `autoStartup` status. Also, we need to call the `configureTopology(topology)` method immediately after the topology creation so that some clients, such as the `TopologyTestDriver`, may use a fully configured topology.

Reply:
Also, please add your name as an `author` of this class.

Reply:
Looks like this comment is not addressed? https://github.com/spring-projects/spring-kafka/pull/3172/files#r1546662939

Reply:
@sobychacko, I'm so sorry, i missed this comments. 

> It might be better to add this to the afterPropertiesSet() method instead of here. Doing it here adds more responsibilities to the isAutoStartup() method, which should technically only return the current autoStartup status. Also, we need to call the configureTopology(topology) method immediately after the topology creation so that some clients, such as the TopologyTestDriver, may use a fully configured topology.

IMHO, `afterPropertiesSet()` is not proper workaround. 
`streamBuilder#stream()` seems to be called  after `StreamsBuilderFactoryBean#afterPropertiesSet()` are executed.  

When `StreamsBuilderFactoryBean#afterPropertiesSet()` is called, no stream declarations are made in the `StreamBuilder` yet. Therefore, executing `StreamsBuilder.build()` at the time of the afterPropertiesSet() call does not create any topology.

Reply:
And i agree that `isAutoStartup()` has much responsibilities in this case.
If we move `configureTopology(topology)` to `isAutoStartup()`, much responsibilities as well. 

However, as i said above, we cannot build `streamsBuilder` properly in scope of `afterPropertiesSet()`. 

I have one more idea. Creating `StreamsBuilder`.
Remove logic to create`StreamsBuilder` and register `StreamsBuilder` as bean separately. But it must involve much changes.

Reply:
@sobychacko May I try writing a PR again in this direction? 
> Remove logic to createStreamsBuilder and register StreamsBuilder as bean separately.

Or do you have any other good ideas? Please let me know. 🙇‍♂️

Reply:
Just commented on your issue in Spring Framework.
Let's see if `SmartInitializingSingleton.afterSingletonsInstantiated()` implementation help us somehow!

Reply:
@artembilan Hi! 
Thanks your comment. `SmartInitializingSingleton.afterSingletonsInstantiated()` works well. 
I modified code to use `SmartInitializingSingleton.afterSingletonsInstantiated()` and test code as well. 

Please Take a look. 🙇‍♂️ 

CC. @sobychacko